src/encoded/types/patient.py

```diff
@@ -496,7 +506,35 @@ def diagnosis_date(self, request, surgery, radiation, medication):
             if len(non_nephrectomy_dates) > 0:
                 non_nephrectomy_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
                 diagnosis_date = non_nephrectomy_dates[0]
-        return diagnosis_date
+        age_range = "Unknown"
+        ageString = "Unknown"
+        if diagnosis_date != "Not available":
+            birth_date = datetime.strptime("1800-01-01", "%Y-%m-%d")
+            end_date = datetime.strptime(diagnosis_date, "%Y-%m-%d")
+            age = end_date.year - birth_date.year -  ((end_date.month, end_date.day) < (birth_date.month, birth_date.day))
+            ageString = str(age) 
+            if age >= 90:
+                ageString = "90 or above"
+
+
+            if age >= 80:
+                age_range = "80+"
```

Comment:
Should we remove age>=90 and add a range for 80-90? So >=90 won't show in our website.

Reply:
Hold off on this change until I update the years.